src/Avalonia.Controls/Automation/Peers/ControlAutomationPeer.cs

```diff
@@ -116,7 +116,17 @@ protected override IReadOnlyList<AutomationPeer> GetOrCreateChildrenCore()
 
             return result;
         }
+        protected override string? GetHelpTextCore()
+        {          
+            var result = AutomationProperties.GetHelpText(Owner);
 
+            if (string.IsNullOrWhiteSpace(result))
+            {
+                result = ToolTip.GetTip(Owner)?.ToString();
```

Comment:
I'm no expert on automation, but having a complex content (e.g. a control) as a tip will likely read the type's name, which won't help the user. We might want to simply use `as string` here.

Reply:
I don't think we need to automate this at all. We can introduce an attached property that is typed as a string. The property needs to be explicitly set to provide a help text.

Reply:
`AutomationProperties.GetHelpText(Owner)`

Reply:
Indeed, even better since we already have the needed property :)

Reply:
> I'm no expert on automation, but having a complex content (e.g. a control) as a tip will likely read the type's name, which won't help the user. We might want to simply use `as string` here.

good point, I'll do some tests and try as string

Reply:
That Tooltip is most likely not providing a meaningful string. There is no way to validate this. So a fallback like this will produce undesired results.

Reply:
Maybe I am missing something.  Having changed to using `as string` surely any tooltip used will be meaningful, why would a developer have non meaningful text in a text tooltip?  Those who care about accessibility or are legally required to can test and tweak with explicit use of `HelpText`, those that don't care will still end up with more accessible apps.

Reply:
Just checked in WPF and it reads the tooltip if `AutomationProperties.HelpText` is not set.  However if this is a contentious point I could remove the tooltip fall-back for now and start a discussion on accessibility.

Reply:
That change works for me. Thanks.